prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -79,6 +95,23 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
       }
     });
 
+    const enrolleeId = this.enrolment.id;
+
+    this.enrolleeHealthAuthorities.clear();
+    // Any checked HA is converted into an enrollee health authority FormGroup,
+    // which is used to create the payload to back-end
+    this.availableHAs?.controls.forEach((checkbox, i) => {
```

Comment:
Is it possible that `availableHealthAuthorities` would be `null` or `undefined` instead of an empty `FormArray`?

Reply:
Also, I'm not sure you need to do this on submission.  Transforming the data should occur in the `FormStateService` in the `json` method.  That's where you can get the `form.value` and loop over it and transform it for submission or any other use outside the form.  That way all form models and logic are only manipulated through the form state service.  An example, would be `convertSelfDeclarationsToJson` or `convertRemoteAccessSitesToJson` in the `EnrolmentFormStateService`.